interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -287,6 +354,36 @@ func (pdb *PromptDB) AddOrMerge(metadata *prompting.Metadata, path string, reque
 			ids: make(map[prompting.IDType]int),
 		}
 		userEntry = pdb.perUser[metadata.User]
+		userEntry.expirationTimer = time.AfterFunc(initialTimeout, func() {
+			pdb.mutex.Lock()
+			if pdb.maxIDMmap.IsClosed() {
+				return
+			}
+			// Restart expiration timer while holding the lock, so we don't
+			// overwrite a newly-set activity timeout with an initial timeout.
+			// With the lock held, no activity can occur, so no activity timeout
+			// can be set.
+			if userEntry.expirationTimer.Reset(initialTimeout) {
```

Comment:
what if there's already a timer attached there?

Reply:
Per the documentation here https://pkg.go.dev/time#Timer.Reset:
> For a func-based timer created with [AfterFunc](https://pkg.go.dev/time#AfterFunc)(d, f), Reset either reschedules when f will run, in which case Reset returns true, or schedules f to run again, in which case it returns false.

If there's already a timer attached (because the timer expired while some activity-causing query held the lock but had not yet reset the timer), our resetting it to `initialTimeout` will return `true`, so we realize this race occurred and we reset to `activityTimeout` instead. I believe the only lasting side effect is that the expiration timeout gets pushed back a few milliseconds later than it otherwise would have been, because of the time between when we initially override the newly-set `activityTimeout` and then quickly re-set it to `activityTimeout` again when we realize it was previously set.

Reply:
but this line "userEntry.expirationTimer = time.AfterFunc ..." is stomping on a possibly already attached timer, that's what my question is about?